pkg/solana/fees/block_history.go

```diff
@@ -16,6 +16,26 @@ import (
 
 var _ Estimator = &blockHistoryEstimator{}
 
+type EstimationMethod int
+
+const (
+	LatestBlockEstimator EstimationMethod = iota
+	MultipleBlocksEstimator
+)
+
+const MAX_BLOCK_HISTORY_DEPTH = 20
+
+func (em EstimationMethod) String() string {
+	switch em {
+	case LatestBlockEstimator:
+		return "LatestBlock"
+	case MultipleBlocksEstimator:
+		return "MultipleBlocks"
+	default:
+		return "UnknownMethod"
+	}
+}
```

Comment:
I don't think we need to introduce an estimation method here. I think we can infer the same thing with using the `BlockHistorySize` config. Like if `BlockHistorySize=1` then we should do the latest block logic, otherwise do the multiple blocks logic